src/ext/ddtrace.c

```diff
@@ -92,6 +99,39 @@ static void register_span_data_ce(TSRMLS_D) {
     zend_declare_property_null(ddtrace_ce_span_data, "metrics", sizeof("metrics") - 1, ZEND_ACC_PUBLIC TSRMLS_CC);
 }
 
+// true globals; only modify in MINIT/MSHUTDOWN
+static stack_t ddtrace_altstack;
+static struct sigaction ddtrace_sigaction;
+
+#define METRICS_CONST_TAGS "lang:php,lang_version:" PHP_VERSION ",tracer_version:" PHP_DDTRACE_VERSION
+
+static void ddtrace_sigsegv_handler(int sig) {
+    TSRMLS_FETCH();
+    char *dogstatsd_host = get_dd_agent_host();
+    char *dogstatsd_port = get_dd_dogstatsd_port();
+
+    ddtrace_log_errf("Segmentation fault");
+
+    // todo: extract buffer, host, port, and client to module global, so this is just a send
```

Comment:
Is this something we can do for this PR or should we wait and tackle later?

Reply:
We can do this now, or later. I just didn't want to stall an important feature on this extraction when at the moment it will have no effect on code whatsoever until we have other metrics.